### PR TITLE
2. As a user, I want to read all of my shopping lists and the items in them.

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-66-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by eonflower**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

In our Home component, we imported the SingleList component and then mapped through it so that we would get an individual component per everything in the list. Then as props, we added key, name, and path with the correct path in the data being pulled. We removed the setListPath from being passed into the home component because it is a function and would not properly work when used as a prop.

In the SingleList component, we added in the useStateWithStorage function to properly set the list path and to make the functionality of the handleClick function work. We also added a key prop to the list item.

In the List component we also mapped through the data to create separate ListItem components and added the key and name prop. In the ListItem component we also added a key prop.

The only thing worth noting is that there were some hiccups attempting to render the proper data before realizing there was an error in the schema. Once resolving that, we were golden. 

## Related Issue

closes #2 

## Acceptance Criteria

- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Type of Changes

Enhancement

### Before
 We forgot to take screenshots before, so we logged out to show what it would look like.

Screenshot 2024-02-07 at 10 56 02 AM

Screenshot 2024-02-07 at 10 56 36 AM

### After

Screenshot 2024-02-07 at 10 54 03 AM

Screenshot 2024-02-07 at 10 54 25 AM

## Testing Steps / QA Criteria

1. From the code, type "npm start" in your terminal
2. Click the Sign In button and sign in using the user you have connected to firebase
3. Click the Groceries button
4. Add /list to the url (should look like localhost:3000/list)
5. View the list :)

### CODE REVIEW

File: src/components/SingleList.jsx [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+	);
 
-export function SingleList({ name, path, setListPath }) {
 	function handleClick() {
 		setListPath(path);
 	}
 
 	return (
-		<li className="SingleList">
+		<li className="SingleList" key={index}>
```
Comment (hun-ah) [inline src/components/SingleList.jsx]: Does the li element need a key prop if there is already a key prop on the SingleList component rendered in Home.jsx?

File: src/components/ListItem.jsx [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, index }) {
+	return (
+		<li className="ListItem" key={index}>
```
Comment (hun-ah) [inline src/components/ListItem.jsx]: Does the li element need a key prop if there is already a key prop on the ListItem component rendered in List.jsx?

File: src/components/SingleList.jsx [hunk 3]
```diff
@@ -1,12 +1,18 @@
 import './SingleList.css';
+import { useStateWithStorage } from '../utils';
+
+export function SingleList({ name, index, path }) {
+	const [listPath, setListPath] = useStateWithStorage(
```
Comment (hun-ah) [inline src/components/SingleList.jsx]: If this is already used in App.jsx, does it need to be initialized again here? I think setListPath can be passed down from Home into this component. We tested it and it seemed to work okay.

**Review by hun-ah — COMMENTED** [review]
Looks really great overall! See comments, particularly one about useStateWithStorage being initialized in SingleList.jsx (it is already initialized in App.jsx and setListPath can be passed down through the other components successfully based on our testing).

Also, see this from React Docs Re: key prop, particularly 'Why does React need keys' section: https://react.dev/learn/rendering-lists. It seems like we might need something more unique than the index - what do you think?

File: src/components/SingleList.jsx (hunk 3 above)
Comment (eonflower) [reply src/components/SingleList.jsx:5]: Can you show me what you did while testing to get it to work without initializing again in the SingleList component?

Comment (eonflower) [reply src/components/SingleList.jsx]: You're right! Got it working after some tinkering.

**Review by jeremiahfallin — APPROVED** [review]
Nice work, looks good to me! Nice catch removing the unnecessary comments. Feel free to merge whenever you're ready.

**Review by StefieCaff — APPROVED** [review]
Great job ya'll. Looks like the key will be more unique, is is now being assigned only once, and the setListPath is being passed through the prop chain.

**Review by hun-ah — APPROVED** [review]

**Pull request merged**